azure/services/scalesetvms/spec.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scalesetvms
+
+import (
+	"context"
+)
+
+// ScaleSetVMSpec defines the specification for a VMSS VM.
+type ScaleSetVMSpec struct {
+	Name          string
+	InstanceID    string
+	ResourceGroup string
+	ScaleSetName  string
+	ProviderID    string
+	ResourceID    string
+	IsFlex        bool
+}
+
+// ResourceName returns the name of the VMSS VM.
+func (s *ScaleSetVMSpec) ResourceName() string {
+	return s.InstanceID
```

Comment:
@CecileRobertMichon I ran into a bit of an awkward situation trying to get this to work. Although we have a `Name` field, the uniform VMSS instances are identified by the `InstanceID`, not the `Name` field. WDYT about this work around?

Reply:
I think this is fine, however can we add a code comment to explain?